packages/bart/src/index.ts

```diff
@@ -0,0 +1,626 @@
+           }
+         },
+         choices: [trial_text.pump_button, trial_text.collect_button],
+         trial_duration: enable_timeout ? trial_timeout : null,
+         button_html: (choice: string) => {
+              if (choice === trial_text.pump_button) {
+                 return `<button class="jspsych-btn continue-button">${choice}</button>`;
+              } else {
+                 return `<button class="jspsych-btn continue-button">${choice}</button>`;
+              }
+         },
+         on_finish: (data: any) => {
+           if (data.response === 0) {
+             pump_count++;
+             if (pump_count >= explosion_point) {
+               balloon_popped = true;
+             }
+           } else if (data.response === 1) {
+             cashed_out = true;
+           } else if (data.response === null && enable_timeout) {
+             // Timeout occurred - auto collect
+             cashed_out = true;
+             timed_out = true;
+           }
+         }
+       }],
+       loop_function: () => !balloon_popped && !cashed_out
+     };
+
+               const outcome = {
+       type: HtmlButtonResponsePlugin,
+       stimulus: () => {
+         const style = getBalloonStyle(pump_count);
+         if (balloon_popped) {
+           return `
+<div class="trial-container">
+                <div class="bart-container">
+               <div class="balloon-area">
+                 <img src="images/transparent_popped_balloon.png" style="${style}" />
+               </div>
+             </div>
+               <div class="task-instructions">
+                 <p><strong>POP!</strong> ${trial_text.balloon_popped_message}</p>
+                 <p>${trial_text.total_earnings_message} <strong>${USDollar.format(.01* jsPsych.data.get().filter({ task: 'bart', exploded: false, cashed_out: true }).select('pump_count').sum())}</strong></p>
+               </div>
+             </div>
+           `;
+         } else {
+           const total_points = pump_count + jsPsych.data.get().filter({ task: 'bart', exploded: false, cashed_out: true }).select('pump_count').sum();
+           const total_money = total_points * currency_unit_per_pump * 0.01;
+           
+           const timeoutMessage = timed_out ? `<p><em>${trial_text.timeout_message}</em></p>` : '';
+
+           return `
+           <div class="trial-container">
+             <div class="task-instructions">
+             <p>${trial_text.collected_message} <strong>${USDollar.format(.01*pump_count)}</strong> this round.</p>
+             ${timeoutMessage}
+             <p>${trial_text.total_earnings_message} <strong>${USDollar.format(total_money)}</strong></p>
+             </div>
+           </div>
+           `;
+         }
+       },
+       choices: [trial_text.continue_button],
+       button_html: (choice: string) => `<button class="jspsych-btn continue-button">${choice}</button>`,
+       on_finish: (data: any) => {
+         data.task = 'bart';
```

Comment:
here too. the params that are fixed strings can go in the data object for the trial. the ones that are dependent on variables need to go here.